dynamiqs/time_tensor.py

```diff
@@ -0,0 +1,257 @@
+from __future__ import annotations
+
+from abc import abstractmethod, abstractproperty
+from typing import get_args
+
+import torch
+from torch import Tensor
+
+from ._utils import obj_type_str, type_str
+from .utils.tensor_types import ArrayLike, Number, get_cdtype, to_device, to_tensor
+
+__all__ = ['tt']
+
+
+def to_time_tensor(
+    x: ArrayLike | callable[[float], Tensor],
+    *,
+    dtype: torch.dtype | None = None,
+    device: str | torch.device | None = None,
+) -> TimeTensor:
+    dtype = get_cdtype(dtype) if dtype is None else dtype  # assume complex by default
```

Comment:
```suggestion
    dtype = dtype or get_cdtype(None)
```

Reply:
Arf, I'm not fan of this construct but ok for here (it's very risky whenever the argument may be a list, string, or any object that may evaluate to False somehow).